neuro_san/internals/graph/activations/branch_activation.py

```diff
@@ -197,8 +198,5 @@ async def use_tool(self, tool_name: str, tool_args: Dict[str, Any], sly_data: Di
             # Nope. Just a regular http connection failure given the tool_name. Can't help ya.
             raise exception
 
-        # We got a list of messages back as a string. Take the last.
-        message_dict: Dict[str, Any] = message_list[-1]
-        content: str = message_dict.get("content")
-
-        return content
+        # We got a message back, take the content as the return string
+        return message.content
```

Comment:
This use_tool() is a way for CodedTools to call back out to the agent graph.
It's intended to provide the text output of a CodedTool, so its return value is a string.